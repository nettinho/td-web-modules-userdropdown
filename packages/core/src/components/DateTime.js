import React from "react";
import PropTypes from "prop-types";
import { FormattedDate } from "react-intl";

export const DateTime = ({ value }) => (
  <FormattedDate
    value={new Date(value)}
    year="numeric"
    month="short"
    day="2-digit"
    hourCycle="h23"
    hour="2-digit"
    minute="2-digit"
    second="2-digit"
  />
);

DateTime.propTypes = {
  value: PropTypes.string
};

export default DateTime;
