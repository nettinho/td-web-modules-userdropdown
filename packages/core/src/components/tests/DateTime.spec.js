import React from "react";
import { shallow } from "enzyme";
import { DateTime } from "../DateTime";

describe("<DateTime value='123124' />", () => {
  it("matches the latest snapshot", () => {
    const wrapper = shallow(<DateTime value="2018-06-27T07:32:53.154377Z" />);
    expect(wrapper).toMatchSnapshot();
  });
});
