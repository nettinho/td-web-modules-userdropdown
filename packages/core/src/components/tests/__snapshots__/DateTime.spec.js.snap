// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DateTime value='123124' /> matches the latest snapshot 1`] = `
<FormattedDate
  day="2-digit"
  hour="2-digit"
  hourCycle="h23"
  minute="2-digit"
  month="short"
  second="2-digit"
  value={2018-06-27T07:32:53.154Z}
  year="numeric"
/>
`;
